keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -197,11 +198,9 @@ class CombinedRepository(
 
   private fun Set<DeliveryArtifact>.transform(deliveryConfigName: String) =
     map { artifact ->
-      when (artifact) {
-        is DockerArtifact -> artifact.copy(deliveryConfigName = deliveryConfigName)
-        is DebianArtifact -> artifact.copy(deliveryConfigName = deliveryConfigName)
-        else -> error("Unrecognized artifact type: ${artifact.type}")
-      }
+      val artifacAsMap: MutableMap<String, Any?> = objectMapper.convertValue(artifact)
+      artifacAsMap["deliveryConfigName"] = deliveryConfigName
+      objectMapper.convertValue(artifacAsMap, artifact.javaClass)
```

Comment:
Is this necessary because we no longer guarantee that all artifacts are data classes?

Reply:
Exactly. It's not super pretty, but we can't make assumptions about how plugin developers will implement sub-classes of `DeliveryArtifact`.